File: 1047.rs
use std::io;

fn main() {
  let mut input = String::new();

  io::stdin().read_line(&mut input).unwrap();

  let values: Vec<&str> = input.split_whitespace().collect();

  let h1 = values[0].parse::<i16>().unwrap();
  let m1 = values[1].parse::<i16>().unwrap();

  let h2 = values[2].parse::<i16>().unwrap();
  let m2 = values[3].parse::<i16>().unwrap();

  let t1 = h1 * 60 + m1;
  let t2 = h2 * 60 + m2;

  let t = if t2 > t1 { t2 - t1 } else { t1 - t2 + 24 * 60 };

  let hours = t / 60;
  let minutes = t % 60;

  println!("O JOGO DUROU {} HORA(S) E {} MINUTO(S)", hours, minutes);
}

File: Iniciante/1047/1047.java
import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);

    String[] values = scan.nextLine().split(" ");

    int h1 = Integer.parseInt(values[0]);
    int m1 = Integer.parseInt(values[1]);

    int h2 = Integer.parseInt(values[2]);
    int m2 = Integer.parseInt(values[3]);

    int t1InMinutes = h1 * 60 + m1;
    int t2InMinutes = h2 * 60 + m2;

    int diff = t2InMinutes - t1InMinutes;

    if (t1InMinutes >= t2InMinutes) {
      diff = 24 * 60 + diff;
    }

    int hours = diff / 60;
    int minutes = diff % 60;

    System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", hours, minutes);
  }
}
